packages/vscode/src/index.ts

```diff
@@ -29,6 +29,9 @@ export const middleware: lsp.Middleware = {
 			if (action.command) {
 				action.command = parseServerCommand(action.command);
 			}
+			if (action.edit) {
+				normalizeCodeActionEdit(document, action);
+			}
```

Comment:
The way servers like `typescript-language-server` handle that is they ensure that `Configure` request is always sent for the current file before getting code actions or doing other relevant requests. Is that not possible here?

Line endings is just one of the aspects. There are also other formatting choices like whether to use spaces or not, whether to use semicolons etc. So this doesn't seem like a very comprehensive solution.

Reply:
@rchl I'm quite confused.

> Line endings is just one of the aspects.

Since the package is intended for vscode only, we should never care about it, thats why it is [always set to `\n`](https://github.com/zardoy/plugins/blob/9b5632d35bd329ce5dccb19fd1071395684f40d7/packages/typescript/src/configs/getFormatCodeSettings.ts#L22).
Other format options are [already used](https://github.com/zardoy/plugins/blob/master/packages/typescript/src/configs/getFormatCodeSettings.ts), the only issue was `formatOptions` that contains indentation information

`normalizeCodeActionEdit` only changes indentation, I'm replicating vscode behavior: https://github.com/microsoft/vscode/blob/3059063b805ed0ac10a6d9539e213386bfcfb852/extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts#L59-L62

But you're right ideally it should be passed during request, but as I understood from https://github.com/vuejs/language-tools/pull/2498#issuecomment-1466170104 its currently not possible / no easy way to do it. Though, AFAICS from my use cases the current solution works ideally

Reply:
I was indeed confused about line breaks vs. indentation.

But the same point still stands that indentation style is just one of the options passed through the `Configure` request so ideally that request should be made.

I don't know code base well enough to say whether it's feasible to do so from this point I'm just deferring to @johnsoncodehk to give his take and decide whether that's fine.

Reply:
I hope to avoid use middleware handling it, which causes other language clients to have to follow if they want the same functionality. I think it would be best to raise this requirement to the LSP repo. Also `replaceAll()` may break `\t` in TemplateLiteral.

If this feature is important, these two approaches may work:

1. Infer FormattingOptions from the upper and lower codes, I think there will be ready-made solutions.
2. Write the FormattingOptions of the formatting request to `tmpdir()` or similar to share it.

Reply:
> 2\. Write the FormattingOptions of the formatting request to `tmpdir()` or similar to share it.

Isn't creating custom request such as sync format options much easier?

Though it is much easier to go with first way, so, let's infer code-style in place.

Reply:
Because custom request requires downstream language clients to implement, I hope to reduce the necessary work for the language clients.

Reply:
> Though it is much easier to go with the first way, so, let's infer code-style in place.

Done here: https://github.com/volarjs/plugins/pull/30/commits/65c2a15c70a8a42e02a92334e4888101e93f5298 so I'm closing this pr